### PR TITLE
Feature: Admin v2 Announcement Details Page

### PR DESCRIPTION
**Pull request by KevinMulhern**
Because:
- As an admin, I want to be able to view, create, edit and delete announcements.

This commit:
- Adds view, create, update and delete Crud actions and views for announcements
- Adds rails date field to our tailwind form builder
- Adds new validations for announcement learn more url and expires at attributes

### CODE REVIEW

**Comment by Asartea** [comment]
Looks great! 🚀 

Played around with it a bit, and a few notes:
- I can dismiss the preview, but I can't figure out a way to then get it to show up again.
- It isn't possible to create an announcement that expires in the past, but it is possible to create one that expires today, which causes it to expire immediately. It might make more sense if the check is either "date is in the future" or the expired check is "date is in the past"
- It might be nice to be able to actually mark an announcement as expired, for cases where you want it to stop showing up but still want to preserve it in the expired announcements section

(Also: out of the scope of this PR, but on the index "Announcements displayed to everyone on the site as a banner." > "Announcements are displayed to everyone on the site as a banner."

**Comment by KevinMulhern** [comment]
Thanks for the feedback @Asartea, great stuff!

> It might be nice to be able to actually mark an announcement as expired, for cases where you want it to stop showing

This is a pretty glaring omission when you point it out haha. I agree it would be very nice to have. I think we'd need to switch to a status instead of relying on the expiry date to do it correctly. That edges it ever so slightly out of scope of this. But I'll put a story together for it.

I've gone ahead and put in the rest of your suggestions though. Thanks again!

**Comment by KevinMulhern** [comment]
Steps for QA

1. Log into admin v2: admin@odin.com, password123

Creating a new announcement:
1. Click "Announcements" on the side nav
2. Click "New announcement" button
3. Fill in the form and submit
4. The new announcement
5. Visit the home page /home
6. The announcement should be displayed as a a banner

Editing announcement
1. Go back to the announcement you just created - click view from the all announcements page
2. Click the "Edit" button
3. Change the message to: "Edited message" and click save
5. Edited message" should be displayed on the homepage banner

Deleting announcement
1. Go back to the announcement you created once again - click view from the all announcements page
2. Click the "Delete" button
3. The announcements should be removed from the all announcements page
4. The banner should no longer be displayed on the homepage

File: spec/system/admin_v2/announcements_spec.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      click_link('Delete')
+
+      expect(page).to have_current_path(admin_v2_announcements_path)
+      expect(page).not_to have_content('Test Message')
+
+      using_session('learner') do
+        sign_in(create(:user))
+
+        visit home_path
+        expect(page).not_to have_content('Edited test Message')
```
Comment (Asartea) [inline spec/system/admin_v2/announcements_spec.rb:66]: Shouldn't this be `Test Message`?

```suggestion
        expect(page).not_to have_content('Test Message')
```

**Comment by zachmmeyer** [comment]
@KevinMulhern I missed the deployed ODE for this. I can test the functionality when it is available again.

**Comment by Asartea** [comment]
Played around with it when the deployment was active, and can confirm all the QA steps checked out

File: spec/system/admin_v2/announcements_spec.rb (hunk 1 above)
Comment (KevinMulhern) [reply spec/system/admin_v2/announcements_spec.rb]: Nice spot @Asartea, thanks!

**Comment by KevinMulhern** [comment]
Hey @zachmmeyer, its back up on a review app!

**Comment by zachmmeyer** [comment]
Thank you @KevinMulhern !
✅ Test announcement with working hyperlink.
✅ Announcement is persistent on all pages unless cleared with close button
image
✅ Text and link change when announcement is edited.
image
❌ The message box did not like the entire work of Hamlet. Would be cool to see a character limit.
image
✅ Making an announcement and then deleting it deletes the announcement, no longer displaying it.

I'm not sure how long the deployment is live for but I can set the expiration date to the next day and see if it expires, but that's not part of the AC.

**Review by zachmmeyer — APPROVED** [review]
Meant to approve. I have an announcement waiting for the 25th but I'm not sure if this will be up that long.

**Comment by KevinMulhern** [comment]
Thanks @zachmmeyer, character limit is a great spot! I've put in a limit of 100 characters - that should hopefully be all we need with the learn more url available.

Review apps are up for 1 day, I can increase that. But I wouldn't worry about QA'ing the expiry, this only adds a new interface for managing announcements to replace our existing one. The code for displaying/hiding expired announcements hasn't changed and been there for years. It's well battle tested 💪

**Comment by zachmmeyer** [comment]
Then it sounds like we're good to go!
image

**Pull request merged**